test/Common/DurableTaskEndToEndTests.cs

```diff
@@ -685,85 +636,71 @@ public async Task OutputsValidJSONLogs()
+                predicate: () => File.Exists(LinuxAppServiceLogger.LoggingPath),
+                conditionDescription: "Log file exists",
+                timeout: TimeSpan.FromSeconds(30));
 
-            // Validating JSON outputs
-            foreach (string line in lines)
-            {
-                // (1) Ensure it can be parsed to JSON
-                JObject json = JObject.Parse(line);
+            await TestHelpers.WaitUntilTrue(
+                predicate: () =>
+                {
+                    bool foundEtwEventSourceLog, foundAzureStorageLog, foundDurableTaskCoreLog, foundActivityId, foundRelatedActivityId;
+                    foundEtwEventSourceLog = foundAzureStorageLog = foundDurableTaskCoreLog = foundActivityId = foundRelatedActivityId = false;
 
-                // (2) Contains minimal expected fields
-                List<string> keys = json.Properties().Select(p => p.Name).ToList();
-                Assert.Contains("EventStampName", keys);
-                Assert.Contains("EventPrimaryStampName", keys);
-                Assert.Contains("ProviderName", keys);
-                Assert.Contains("TaskName", keys);
-                Assert.Contains("EventId", keys);
-                Assert.Contains("TimeStamp", keys);
-                Assert.Contains("Tenant", keys);
-                Assert.Contains("Pid", keys);
-                Assert.Contains("Tid", keys);
+                    string[] lines = TestHelpers.WriteSafeReadAllLines(LinuxAppServiceLogger.LoggingPath);
 
-                // recording EventSource providers seen
-                int eventId = (int)json.GetValue("EventId");
-                if (eventId == 202)
-                {
-                    foundEtwEventSourceLog = true;
-                }
-                else if (eventId == 10)
-                {
-                    foundDurableTaskCoreLog = true;
-                }
-                else if (eventId == 120)
-                {
-                    foundAzureStorageLog = true;
-                }
+                    // Validating JSON outputs
+                    foreach (string line in lines)
+                    {
+                        // (1) Ensure it can be parsed to JSON
+                        JObject json = JObject.Parse(line);
 
-                // recording if ActivityId and RelatedActivityId are seen
-                // we expect to see them, at some point, in a trivial orchestrator
-                if (keys.Contains("ActivityId"))
-                {
-                    foundActivityId = true;
-                }
+                        // (2) Contains minimal expected fields
+                        TestHelpers.IsValidJSONLog(json);
 
-                if (keys.Contains("RelatedActivityId"))
-                {
-                    foundRelatedActivityId = true;
-                }
+                        // recording EventSource providers seen
+                        int eventId = (int)json.GetValue("EventId");

```

Comment:
Is EventId the best way of identifying provider? I get that these event ids are unique across providers (at least I believe so), but there should be a Provider field that is more conclusive right?